### PR TITLE
drivers: pinmux: stm32h7: fix incorrect USART6 pin alternate functions

### PR DESCRIPTION
**Pull request by lochej**
Fixed alternate function of PG9/PG14 for USART6 with respect to the datasheet (p98-p99 stm32h743zi.pdf mcu datasheet)
Must be AF7 instead of AF8.

Validated on STM32H743ZI:
Enables the USART6 to be used on D0/D1 pins of the Nucleo H743ZI board.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

### CODE REVIEW

**Review by erwango — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Comment by erwango** [comment]
@lochej Thanks for this contribution.
One comment though (but you don't need to modify anything):
In general, the rule is that people add their copyright only for major contribution on the file, which is not exactly the case here, but this will be the case with your ethernet driver proposal ;-).

**Comment by lochej** [comment]
> @lochej Thanks for this contribution.
> One comment though (but you don't need to modify anything):
> In general, the rule is that people add their copyright only for major contribution on the file, which is not exactly the case here, but this will be the case with your ethernet driver proposal ;-).

You are absolutely right that sounds fair. I did this small PR get more familiar with the git contribution flow :) It's OK if my name doesn't show up in this file ^^

File: drivers/pinmux/stm32/pinmux_stm32h7.h [hunk 1]
```diff
@@ -507,7 +508,7 @@
 	(STM32_PINMUX_ALT_FUNC_8 | STM32_PUSHPULL_PULLUP)
```
Comment (FRASTM) [inline drivers/pinmux/stm32/pinmux_stm32h7.h:508]: why then leaving USART6_RTS and CTS wrongly on  AF8,  AF7 is the correct setting.

Idem for PG15 below

Comment (lochej) [reply drivers/pinmux/stm32/pinmux_stm32h7.h]: You are absolutely right ! I didn't have a look at the other pins as I didn't test them. I have submitted a new commit with all of the USART6 AF corrections. :+1 Thanks for your feedback

**Review by carlescufi — APPROVED** [review]

**Review by FRASTM — APPROVED** [review]

**Comment by ioannisg** [comment]
@lochej one more hint from me; you could just amend (commit --amend) the first patch and remove the copyright adding, so you don't have to do it and undo it in the following commit.

**Comment by lochej** [comment]
> @lochej one more hint from me; you could just amend (commit --amend) the first patch and remove the copyright adding, so you don't have to do it and undo it in the following commit.

Thank you for your advice @ioannisg ! I have done the changes based on your recommendations. I'm still in the process of learning git and I'm getting more comfortable every day :smile:

**Pull request merged**